mlflow/models/evaluation/base.py

```diff
@@ -1643,14 +1643,32 @@ def pred_sample(eval_df, _builtin_metrics, _artifacts_dir):
 
     _EnvManager.validate(env_manager)
 
+    # If Dataset is provided, the targets and predictions can only be specified by the Dataset,
+    # not the targets and predictions parameters of the mlflow.evaluate() API.
+    if isinstance(data, Dataset) and targets is not None:
```

Comment:
Can we check if the dataset specifies `targets`? If it does, can we adjust the error message accordingly? I.e. if dataset doesn't have targets, tell users how to set them (give an example). If the dataset does have targets, tell them to remove the targets argument from `mlflow.evaluate()`.

This way, the user doesn't have

Reply:
I updated the error message to include both specifying the targets in the dataset.targets and using targets=None in the top-level parameter.
With this update, the user no longer need to call `mlflow.evaluate()` twice before figuring out what to do.
I think it's less confusing to have a single error message that describes what is needed.

Wdyt?

Reply:
Looks great!